### PR TITLE
Log additional performance metrics in `create_snapshots_reports_scorecard.py`

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds additional performance metrics to the logging of `extras/create_snapshots_reports_scorecard.py`:
 * Maximum, median, mean, minimum snapshot-generation duration
 * Maximum, median, mean, minimum report-generation duration

I also took this opportunity to relocate some existing logging (top-10 longest snapshots and reports) to the end of the script, where it can be more easily parsed along with the new output listed above (see https://github.com/cisagov/cyhy-reports/commit/965dca651f6d7d03f032519b333dc1c7a06bbee2).

## 💭 Motivation and context ##

Having this additional information will be useful as we adjust our parallelization (thread count) settings in order to find our optimal configuration.

## 🧪 Testing ##

I stubbed out any non-relevant code in `create_snapshots_reports_scorecard.py` and used a `sleep` command (with a random duration) instead of generating snapshots and reports.  This allowed me to run the script without affecting any data.  I confirmed that output like this was logged:

```console
2024-03-20 15:40:09,444 INFO - Snapshot performance:
2024-03-20 15:40:09,445 INFO -   Maximum snapshot: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,447 INFO -   Median snapshot: 6.0 seconds (0.1 minutes)
2024-03-20 15:40:09,447 INFO -   Mean snapshot: 6.0 seconds (0.1 minutes)
2024-03-20 15:40:09,448 INFO -   Minimum snapshot: 1.0 seconds (0.0 minutes)
2024-03-20 15:40:09,448 INFO - Longest snapshots:
2024-03-20 15:40:09,448 INFO -   AAA: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,448 INFO -   BBB: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,448 INFO -   CCC: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,448 INFO -   DDD: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO -   EEE: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO -   FFF: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO -   GGG: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO -   HHH: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO -   III: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO -   JJJ: 10.0 seconds (0.2 minutes)
2024-03-20 15:40:09,449 INFO - Report performance:
2024-03-20 15:40:09,450 INFO -   Maximum report: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,450 INFO -   Median report: 4.0 seconds (0.1 minutes)
2024-03-20 15:40:09,450 INFO -   Mean report: 4.2 seconds (0.1 minutes)
2024-03-20 15:40:09,450 INFO -   Minimum report: 1.0 seconds (0.0 minutes)
2024-03-20 15:40:09,450 INFO - Longest reports:
2024-03-20 15:40:09,450 INFO -   LLL: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   MMM: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   NNN: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   OOO: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   PPP: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   QQQ: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   RRR: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   SSS: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   TTT: 7.0 seconds (0.1 minutes)
2024-03-20 15:40:09,451 INFO -   UUU: 7.0 seconds (0.1 minutes)
```

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Deploy these changes to Production.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Review by dv4harr10 — APPROVED** [review]

File: extras/create_snapshots_reports_scorecard.py [hunk 1]
```diff
@@ -1021,6 +1012,39 @@ def main():
                 else:
                     logging.error(i)
 
+        if not args["--no-snapshots"]:
+            logging.info("Snapshot performance:")
+            durations = [x[1] for x in snapshot_durations]
+            max = numpy.max(durations)
+            logging.info("  Maximum snapshot: %.1f seconds (%.1f minutes)", max, max / 60)
```
Comment (mcdonnnj) [inline extras/create_snapshots_reports_scorecard.py]: I'm just not a fan of "Maximum" in the context of runtime. This also matches the language of the 10 longest output.

```suggestion
            logging.info("  Longest snapshot: %.1f seconds (%.1f minutes)", max, max / 60)
```

File: extras/create_snapshots_reports_scorecard.py [hunk 2]
```diff
@@ -1021,6 +1012,39 @@ def main():
...
+            median = numpy.median(durations)
+            logging.info("  Median snapshot: %.1f seconds (%.1f minutes)", median, median / 60)
+            mean = numpy.mean(durations)
+            logging.info("  Mean snapshot: %.1f seconds (%.1f minutes)", mean, mean / 60)
+            min = numpy.min(durations)
+            logging.info("  Minimum snapshot: %.1f seconds (%.1f minutes)", min, min / 60)
```
Comment (mcdonnnj) [inline extras/create_snapshots_reports_scorecard.py]: Same rationale.

```suggestion
            logging.info("  Shortest snapshot: %.1f seconds (%.1f minutes)", min, min / 60)
```

File: extras/create_snapshots_reports_scorecard.py [hunk 3]
```diff
@@ -1021,6 +1012,39 @@ def main():
...
+
+            snapshot_durations.sort(key=lambda tup: tup[1], reverse=True)
+            logging.info("Longest snapshots:")
+            for i in snapshot_durations[:10]:
+                logging.info("  %s: %.1f seconds (%.1f minutes)", i[0], i[1], i[1] / 60)
+
+        logging.info("Report performance:")
+        durations = [x[1] for x in report_durations]
+        max = numpy.max(durations)
+        logging.info("  Maximum report: %.1f seconds (%.1f minutes)", max, max / 60)
```
Comment (mcdonnnj) [inline extras/create_snapshots_reports_scorecard.py]: Same rationale.

```suggestion
        logging.info("  Longest report: %.1f seconds (%.1f minutes)", max, max / 60)
```

File: extras/create_snapshots_reports_scorecard.py [hunk 4]
```diff
@@ -1021,6 +1012,39 @@ def main():
...
+        median = numpy.median(durations)
+        logging.info("  Median report: %.1f seconds (%.1f minutes)", median, median / 60)
+        mean = numpy.mean(durations)
+        logging.info("  Mean report: %.1f seconds (%.1f minutes)", mean, mean / 60)
+        min = numpy.min(durations)
+        logging.info("  Minimum report: %.1f seconds (%.1f minutes)", min, min / 60)
```
Comment (mcdonnnj) [inline extras/create_snapshots_reports_scorecard.py]: Same rationale.

```suggestion
        logging.info("  Shortest report: %.1f seconds (%.1f minutes)", min, min / 60)
```

File: extras/create_snapshots_reports_scorecard.py [hunk 5]
```diff
@@ -15,6 +15,7 @@
 import distutils.dir_util
 import glob
 import logging
+import numpy
```
Comment (mcdonnnj) [inline extras/create_snapshots_reports_scorecard.py]: Please move this to the other non-stdlib imports.

**Review by mcdonnnj — APPROVED** [review]
Makes sense 👍 I do have some suggestions for your consideration.

File: extras/create_snapshots_reports_scorecard.py (hunk 5 above)
Comment (dav3r) [reply extras/create_snapshots_reports_scorecard.py]: Good catch, please see https://github.com/cisagov/cyhy-reports/pull/116/commits/f8396987650c8a65565f1c14ae92f4828dd8a524.

**Comment by dav3r** [comment]
I have manually deployed this change to Production.

**Pull request merged**